### PR TITLE
feat: New id argument to print(), with new option print.id

### PR DESCRIPTION
**Pull request by krlmlr**
With stable snapshot tests, for #466.

### CODE REVIEW

**Comment by krlmlr** [comment]
There's light at the end of the rabbit hole.

This is a rather large change, but the commits make sense individually. I need this to troubleshoot failures in the indexing code.

File: R/par.R [hunk 1]
```diff
@@ -186,10 +190,17 @@ igraph_options <- function(...) {
...
   ## Set them
   names(temp) <- paste0("igraph::", names(temp))
-  do.call(set_config_in, c(temp, list(.in = parent.frame())))
+  do.call(set_config_in, c(temp, list(.in = .in)))
   invisible(old)
 }
 
+local_igraph_options <- function(..., .in = parent.frame()) {
+  old <- igraph_options(..., .in = .in)
+  print(old)
```
Comment (ntamas) [inline R/par.R:199]: Is this `print()` statement necessary here or is it only for debugging purposes?

Comment (krlmlr) [reply R/par.R]: Good catch, removed.

File: R/par.R [hunk 2]
```diff
@@ -185,10 +190,16 @@ igraph_options <- function(...) {
...
   ## Set them
   names(temp) <- paste0("igraph::", names(temp))
-  do.call(set_config_in, c(temp, list(.in = parent.frame())))
+  do.call(set_config_in, c(temp, list(.in = .in)))
   invisible(old)
 }
 
+local_igraph_options <- function(..., .in = parent.frame()) {
+  old <- igraph_options(..., .in = .in)
+  withr::defer(rlang::inject(igraph_options(!!!old)), envir = .in)
```
Comment (ntamas) [inline R/par.R:199]: I guess that we need to move `withr` from the `Suggests:` section of the `DESCRIPTION` file to the `Imports:` section, right?

Comment (krlmlr) [reply R/par.R:199]: Not necessarily. This isn't part of an exported API, and we can realistically expect that it's available during testing.

**Comment by ntamas** [comment]
Merging PR upon CI pass, FYI.

**Pull request merged**